fingerprint.c

```diff
@@ -44,14 +51,32 @@ typedef struct {
     char db_path[255];
     pthread_mutex_t lock;
     uint64_t challenge;
+    int state;
 } sony_fingerprint_device_t;
 
-void *enroll_thread_loop(void *arg)
-{
-    ALOGI("%s", __func__);
-    sony_fingerprint_device_t *sdev = (sony_fingerprint_device_t*)arg;
-    fingerprint_notify_t callback = sdev->device.notify;
 
+
+static int getState(sony_fingerprint_device_t* sdev) {
+    ALOGD("%s", __func__);
+    int state = STATE_IDLE;
+    pthread_mutex_lock(&sdev->lock);
```

Comment:
There's no benefit to locking here - it's an atomic operation to copy the integer.